File: PersonalProjects/prosjekt-3/data/scraper/country.py

from uuid import uuid4

class MovieCountry:
    countries = {}
    def getCountry(self, country):
        if country in MovieCountry.countries:
            return MovieCountry.countries[country]
        else:
            id = MovieCountry.countries[country] = uuid4()
            return id
    

    def getCountries(self, country):
        return [MovieCountry.getCountry(self, l.strip()) for l in country.split(",")]

File: PersonalProjects/prosjekt-3/backend/crud/country.ts
import { Client, QueryResultRow } from "pg";
import { Country } from "./../types";

export const getCountries = async (
  movie: QueryResultRow,
  client: Client
): Promise<Country[]> => {
  const result = await client.query(
    "select c.id, c.name from moviecountry mc inner join country c on c.id = mc.countryid where mc.movieid=$1::uuid ",
    [movie.id]
  );
  return result.rows;
};
